scripts/eslint_rules/no-deprecated-clone.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+
+const isInIncludedFolder = filename => {
+  return /(?:packages|codemods|experimental)\/babel-/.test(filename);
+};
+
+const message =
+  "t.clone() and t.cloneDeep() are deprecated. Use t.cloneNode() instead.";
+
+module.exports = {
+  meta: {
+    schema: [],
+  },
+  fixable: "code",
+  create(context) {
+    if (!isInIncludedFolder(context.getFilename())) {
+      return {};
+    }
+
+    return {
+      CallExpression(node) {
+        const callee = node.callee;
+
+        if (callee.type !== "MemberExpression") return;
+        if (callee.computed) return;
+
+        const object = callee.object;
+        if (object.type !== "Identifier" || object.name !== "t") return;
```

Comment:
i know that this is a convention of ours, but could be more bullet proof with some [`referencesImport`](https://github.com/babel/babel/blob/7064b298d70490b8c129e736e322e0b356e8fb24/packages/babel-traverse/src/path/introspection.js#L160) logic, but probably had to check also for  a destructured argument 🤔

Reply:
Yeah, lots of plugins also take the `types` key from the first argument to the plugin factory

Reply:
Currently https://github.com/babel/babel/pull/7191/files#diff-a22e9508727437cec502b0b6edaaeaf3R16 ensure that it's not used outside of Babel, I think it would be good to allow it.